### PR TITLE
Add round function for decimal types

### PR DESCRIPTION
**Pull request by karteekmurthys**
Round is a Vector function as it requires calculating the result precision at run time.

### CODE REVIEW

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 1]
```diff
@@ -97,6 +97,60 @@ class DecimalBaseFunction : public exec::VectorFunction {
...
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& resultType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    auto rawResults = prepareResults(rows, resultType, context, result);
+    if (args[0]->isConstantEncoding()) {
+      // Fast path for (const, flat).
```
Comment (mbasmanova) [inline velox/functions/prestosql/DecimalArithmetic.cpp:116]: Shouldn't this say "Fast path for constant", since there is only one argument?

Same below.

**Comment by karteekmurthys** [comment]
@mbasmanova @majetideepak addressed review comments. Please review it. I will be adding abs and negate functions on top of this.

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 2]
```diff
@@ -97,6 +97,52 @@ class DecimalBaseFunction : public exec::VectorFunction {
...
+ public:
+  DecimalUnaryBaseFunction(uint8_t aRescale) : aRescale_(aRescale) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& resultType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    auto rawResults = prepareResults(rows, resultType, context, result);
```
Comment (majetideepak) [inline velox/functions/prestosql/DecimalArithmetic.cpp:114]: Add a comment here 
`single-arg deterministic functions receive their only argument as flat or constant only`

**Review by majetideepak — APPROVED** [review]

**Comment by karteekmurthys** [comment]
@Yuhta this PR is the base PR for abs and negate functions. Please review this as well.

**Comment by karteekmurthys** [comment]
Closing this PR as these changes are included in https://github.com/facebookincubator/velox/pull/4623

**Pull request closed without merging**